x/keyshare/keeper/msg_server_override_pub_key.go

```diff
@@ -0,0 +1,111 @@
+package keeper
+
+import (
+	"context"
+	"encoding/json"
+	peptypes "github.com/Fairblock/fairyring/x/pep/types"
+	"strconv"
+
+	"github.com/Fairblock/fairyring/x/keyshare/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// OverrideLatestPubKey updates the public key
+func (k msgServer) OverrideLatestPubKey(goCtx context.Context, msg *types.MsgOverrideLatestPubKey) (*types.MsgOverrideLatestPubKeyResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	params := k.GetParams(ctx)
+
+	trustedAddresses := params.TrustedAddresses
+	if !contains(trustedAddresses, msg.Creator) {
+		return nil, types.ErrAddressNotTrusted.Wrap(msg.Creator)
+	}
+
+	commitments := types.Commitments{
+		Commitments: msg.Commitments,
+	}
+
+	expHeight := params.KeyExpiry + uint64(ctx.BlockHeight())
+
+	if msg.IsPendingPubKey {
+		if _, found := k.GetActivePubKey(ctx); !found {
+			return nil, types.ErrPubKeyNotFound
+		}
+	}
+
+	encryptedKeyShares, err := json.Marshal(msg.EncryptedKeyShares)
+	if err != nil {
+		return nil, err
+	}
+
+	if msg.IsPendingPubKey {
+		k.SetQueuedCommitments(
+			ctx,
+			commitments,
+		)
+		k.SetQueuedPubKey(
+			ctx,
+			types.QueuedPubKey{
+				Creator:            msg.Creator,
+				PublicKey:          msg.PublicKey,
+				Expiry:             expHeight,
+				NumberOfValidators: msg.NumberOfValidators,
+				EncryptedKeyShares: msg.EncryptedKeyShares,
+			},
+		)
+
+		k.pepKeeper.SetQueuedPubKey(
+			ctx,
+			peptypes.QueuedPubKey{
+				Creator:   msg.Creator,
+				PublicKey: msg.PublicKey,
+				Expiry:    expHeight,
+			},
+		)
+	} else {
+		k.SetActiveCommitments(
+			ctx,
+			commitments,
+		)
+
+		k.SetActivePubKey(
```

Comment:
If active pubkey is overwritten and there is already a queued key, then there is a possibility that queued expiry may be less than active expiry.